src/Core/Settings.h

```diff
@@ -80,9 +80,14 @@ class IColumn;
     M(Bool, use_uncompressed_cache, false, "Whether to use the cache of uncompressed blocks.", 0) \
     M(Bool, replace_running_query, false, "Whether the running request should be canceled with the same id as the new one.", 0) \
     M(UInt64, background_buffer_flush_schedule_pool_size, 16, "Number of threads performing background flush for tables with Buffer engine. Only has meaning at server startup.", 0) \
-    M(UInt64, background_pool_size, 16, "Number of threads performing background work for tables (for example, merging in merge tree). Only has meaning at server startup.", 0) \
+    M(UInt64, background_pool_size, 16, "Deprecated. Number of merges that could be performed in background. Has to be not less than `background_merges_pool_size`. Only has meaning at server startup.", 0) \
+    M(Float, background_merges_concurrency_ratio, 3, "Ratio between a number of how many merges could be processed and a number threads to process them. Only has meaning at server startup.", 0) \
+    M(UInt64, background_merges_pool_size, 8, "Number of threads performing background merges for tables. Only has meaning at server startup.", 0) \
```

Comment:
Now I don't like that if we have no mutations, than our pool size is 8 (was 16 before)